src/Router/ModuleRouter.php

```diff
@@ -82,6 +82,21 @@ public function __construct(\Module $module, string $moduledir)
      */
     public function handle(ServerRequestInterface $request) : ResponseInterface
     {
+        $gets = $request->getQueryParams();
+        if (!empty($gets['candID'])) {
+            $request = $request->withAttribute(
+                'Candidate',
+                (\NDB_Factory::singleton())->candidate($gets['candID'])
+            );
+        }
+
+        if (!empty($gets['sessionID'])) {
+            $request = $request->withAttribute(
+                'TimePoint',
+                \TimePoint::singleton($gets['sessionID'])
+            );
+        }
```

Comment:
Does this belong in the module router? Shouldn't it be up to the individual pages/modules whether or not they want to populate these in the request that gets passed to the middleware?

Reply:
like that https://github.com/aces/Loris/pull/3799 ?

Reply:
Yes, I haven't tested it but that looks better to me. (I don't really think those "magic" panels should be in the page decoration middleware at all, it's a relic from main.php. At the minimum they should be a separate middleware that modules need to explicitly add.)